### PR TITLE
Use libovsdb in unidling controller

### PR DESCRIPTION
**Pull request by dave-tucker**
**- What this PR does and why is it needed**

Based on #2217
This switches the `unidling` controller from `fork/exec` of `ovn-sbctl` on a ticker, to setting up a monitor for the `Controller_Event` table, adding any `empty_lb_backends` events to a channel, and processing those events when received.
It also adds a test with a mock southbound db server to show how this can be tested.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/util/libovsdb.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+type ReconnectFunc func(client.Client) error
+
+// newClient creates a new client object given the provided config
+// the stopCh is required to ensure the two goroutines for ssl cert
+// update and reconnect are not leaked
+func newClient(cfg config.OvnAuthConfig, dbModel *model.DBModel, reconnectFn ReconnectFunc, stopCh <-chan struct{}) (client.Client, error) {
```
Comment (jcaamano) [inline go-controller/pkg/util/libovsdb.go:27]: A name like `onReconnectFn` would have helped me understand better what this is for.

**Comment by dave-tucker** [comment]
Now rebased on #2274 but incomplete as there is no way to add an EventHandler in the util function due to the util function not being aware of my channel. This may end up requiring https://github.com/ovn-org/libovsdb/pull/176

**Comment by dave-tucker** [comment]
@dcbw @trozet this is ready to go. Just let me know if you want me to pop that last commit or not.

**Comment by dave-tucker** [comment]
Since #2507 closed, this also carries the fix to only monitor certain SBDB tables, so if CI is green now this is rebased it should 100% be good to go

**Comment by dave-tucker** [comment]
Popped last commit since it was broken out in to #2627

File: go-controller/pkg/ovn/controller/unidling/unidle.go [hunk 2]
```diff
@@ -91,66 +125,81 @@ type ServiceVIPKey struct {
...
+func (uc *unidlingController) DeleteServiceVIPToName(vip string, protocol kapi.Protocol) {
 	uc.serviceVIPToNameLock.Lock()
 	defer uc.serviceVIPToNameLock.Unlock()
 	delete(uc.serviceVIPToName, ServiceVIPKey{vip, protocol})
 }
 
 func (uc *unidlingController) Run(stopCh <-chan struct{}) {
-	ticker := time.NewTicker(5 * time.Second)
-
+	ctx := context.Background()
```
Comment (trozet) [inline go-controller/pkg/ovn/controller/unidling/unidle.go:154]: can you use a context with a the regular ovsdb timeout like we've updated other parts of the code to use?

**Review by trozet — CHANGES_REQUESTED** [review]
looks good @dave-tucker . One minor comment inline. When you push your next version we can make sure it works with the latest libovsdb changes.

**Comment by dcbw** [comment]
Nice cleanup.
/lgtm

**Pull request merged**